test/e2e/framework/node/resource.go

```diff
@@ -67,51 +67,40 @@ func FirstAddress(nodelist *v1.NodeList, addrType v1.NodeAddressType) string {
 	return ""
 }
 
-// TODO: better to change to a easy read name
 func isNodeConditionSetAsExpected(node *v1.Node, conditionType v1.NodeConditionType, wantTrue, silent bool) bool {
 	// Check the node readiness condition (logging all).
+	// Ensure that the condition type and the status matches as desired.
 	for _, cond := range node.Status.Conditions {
-		// Ensure that the condition type and the status matches as desired.
-		if cond.Type == conditionType {
```

Comment:
For line 75 and 77, This looks a little weird.
function IsConditionSetAsExpected and IsConditionSetAsExpectedSilent use isNodeConditionSetAsExpected,
and all values passed to conditionType are "NodeReady"

Reply:
the two different function seem to wrap this one while passing `silent=true/false`.

but the diff here removes the check against conditionType, which makes the name of the function invalid.
i suggest this function is kept like that for now and only try to remove this TODO:

> // TODO: better to change to a easy read name

Reply:
thanks your review. The code implementation here always feels a bit weird i feel.
how about this change now?

Reply:
why modify the contents of the function?

Reply:
see https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/node/resource.go#L75 , 
 if cond.Type == conditionType  is true and cond.Type == v1.NodeReady is true, https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/node/resource.go#L114 never run。
I think it is possible to judge this condition in advance, so move this code. If you don't think need to change it, I can revert it.I can just try to remove this TODO.

Reply:
so in this PR i think we should only:
- removed the TODO `// TODO: better to change to a easy read name`.
- make the functions `is....()` private.

leave this TODO:
> // TODO: check if the Node is tainted once we enable NC notReady/unreachable taints by default

to SIG Node and not touch it.

Reply:
ok

Reply:
done,updated